src/Compilers/Core/Portable/SynthesizedLocalKind.cs

```diff
@@ -230,6 +230,13 @@ internal enum SynthesizedLocalKind
         /// TODO: Avoid using lambdas and display classes for implementation of relaxation stubs and remove this kind.
         /// </summary>
         DelegateRelaxationReceiver = 0x101,
+
+        /// <summary>
+        /// The variable used in async-iterator methods to refer to the CancellationToken
+        /// from `GetAsyncEnumerator(CancellationToken)`.
+        /// Never actually created as a local variable. Its usages are lowered to refer to that parameter instead.
```

Comment:
The uses are *logically* lowered to the parameter but actually lowered to the hoisted field.